src/actions/guides/json_and_apis/cors.cr

```diff
@@ -21,12 +23,109 @@ class Guides::JsonAndApis::Cors < GuideAction
         response.headers["Access-Control-Allow-Origin"] = "*"
         response.headers["Access-Control-Allow-Headers"] =
           "DNT,User-Agent,X-Requested-With,If-Modified-Since,Cache-Control,Content-Type,Range"
+        response.headers["Access-Control-Allow-Methods"] = "*"
         continue
       end
     end
     ```
 
     > Note: the before action method must return `continue`, or a `LuckyWeb::Response`. See [Actions and Routing](#{Guides::HttpAndRouting::RequestAndResponse.path(anchor: Guides::HttpAndRouting::RequestAndResponse::ANCHOR_RUN_CODE_BEFORE_OR_AFTER_ACTIONS_WITH_PIPES)}) for more info.
+
+    ### Complex example
+
+    You may find that you need a little more than a simple setup to handle CORS.
+    Some requests may require a preflight `OPTIONS` call. You can check for that request method.
```

Comment:
How does this work? Do you need to specify an `options` route for every route? It seems like if you didn't it would never hit this `before` action because there'd be no matching requests?

But also I don't know a ton about CORS so maybe I just don't get it :P

Reply:
@russ ^ ??? 😂

Reply:
Bit of a miscommunication there. The example needs to be it's own handler inserted into middleware stack before the RouteHandler. That way CORS is handled for the whole app as opposed to each action.

Reply:
Ok, that makes more sense. I'll move it to a handler